inc/hclib-async.h

```diff
@@ -64,7 +64,9 @@ inline void call_lambda(T* lambda) {
 	const int wid = current_ws()->id;
 	MARK_BUSY(wid);
 	(*lambda)();
-	HC_FREE((void*) lambda);
+    // FIXME - this whole call chain is kind of a mess
```

Comment:
This was just a reminder for me. The issues here are already fixed—but the fixes are further down the pull request pipeline. My vague "mess" description was referring to the following issues:

+ Calls to memory-related C functions from C++ code (malloc/free/memset).
+ Lambdas passed by-value through this call chain, causing a bunch of superfluous copies. (I have an example in a later revision (yet to be pull-requested) showing how each lambda gets copied 2 or 3 times here, vs no copies after some optimization.)
+ Unnecessary levels of indirection around the async and async-await task lambdas.
+ Duplication of code that's implemented in the underlying C API, while it should be easy to delegate to the C code (e.g., for task allocation).

There's still one place in the HCUPC++ code that has a comment like this, but I'm going to talk with Kumar this week (after he has a time to review these PRs) to re-sync the UPC-related code with these changes.